src/Function.cpp

```diff
@@ -681,5 +874,53 @@ bool Function::frozen() const {
     return contents.ptr->frozen;
 }
 
+const map<string, IntrusivePtr<FunctionContents>> &Function::wrappers() const {
+    return contents.ptr->schedule.wrappers();
+}
+
+void Function::add_wrapper(const Function &wrapper, const std::string &f) {
+    contents.ptr->schedule.add_wrapper(wrapper.contents, f);
+}
+
+Function &Function::substitute_calls(const map<Function, Function> &substitutions) {
+    // Replace all calls to functions listed in 'substitutions' with their wrappers.
+    class SubstituteCalls : public IRMutator {
```

Comment:
An inline mutator class caused problems for me on windows. Safer to pull it outside the method and put it in an anonymous namespace instead.